### PR TITLE
Cleanup jfr-streaming

### PR DESCRIPTION
**Pull request by jack-berg**
Changes includes:

- Rename `JfrMetrics` to `JfrTelemetry`. This is looking forward to potentially emitting other telemetry besides metrics in the future.
- `JfrTelemetry` is initialized with `OpenTelemetry` instead of `MeterProvider`. This is a continuation of the last item.
- `JfrTelemetry` implements `Closeable`, and can stop recording.
- All `RecordEventHandler` implementations accept `Meter` as a construction arg. This allows the removal of the awkward post-construction initialization via `initializeMeter`, and the `defaultMeter`.
- Remove some unused constants
- Refactor `AbstractJfrTest` to be a junit5 extension, `JfrExtension`, favoring composition over inheritance. Tests now start and stop recording for each test, improving isolation.

### CODE REVIEW

**Review by breedx-splk — APPROVED** [review]
I left a couple small ideas...but this looks good. This is a big improvement IMO, thanks for this! I think the create-instance-then-later-initialize is a pretty glaring antipattern, so I'm glad we're getting rid of that here.

It's a minor thing, but it does (still?) feel a little weird to me to pass an instance to the handler constructors in a way that is pure side effect. That's a conversation for another time.

File: jfr-streaming/src/testFixtures/java/io/opentelemetry/contrib/jfr/metrics/JfrExtension.java [hunk 1]
```diff
@@ -8,32 +8,35 @@
...
-    GlobalOpenTelemetry.set(OpenTelemetrySdk.builder().setMeterProvider(meterProvider).build());
-    JfrMetrics.enable(meterProvider);
+    jfrTelemetry =
+        JfrTelemetry.create(OpenTelemetrySdk.builder().setMeterProvider(meterProvider).build());
+  }
+
+  @Override
+  public void afterEach(ExtensionContext context) {
+    meterProvider.close();
+    jfrTelemetry.close();
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply jfr-streaming/src/testFixtures/java/io/opentelemetry/contrib/jfr/metrics/JfrExtension.java:39]: Closing the meter provider closes all its readers, so we're safe here.

File: jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/HandlerRegistry.java [hunk 2]
```diff
@@ -31,44 +32,42 @@
...
-  private static final String INSTRUMENTATION_VERSION = "1.7.0-SNAPSHOT";
-
-  private final List<RecordedEventHandler> mappers;
-
-  private HandlerRegistry(List<? extends RecordedEventHandler> mappers) {
-    this.mappers = new ArrayList<>(mappers);
-  }
+  private static final String SCOPE_NAME = "io.opentelemetry.contrib.jfr";
+  // TODO(jack-berg): read from version.properties
+  private static final String SCOPE_VERSION = "1.7.0-SNAPSHOT";
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/HandlerRegistry.java:37]: Bumped it to 1.22.0 for now. I plan on fixing #707 before the next release, if nobody beats me to it.

File: jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/JfrTelemetry.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  private final AtomicBoolean isClosed = new AtomicBoolean();
+  private final RecordingStream recordingStream;
+
+  @SuppressWarnings("CatchingUnchecked")
+  private JfrTelemetry(OpenTelemetry openTelemetry) {
+    List<RecordedEventHandler> handlers = HandlerRegistry.getHandlers(openTelemetry);
+    try {
+      logger.log(Level.INFO, "Starting JfrTelemetry");
+      recordingStream = new RecordingStream();
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/JfrTelemetry.java:30]: Yeah that's a good idea. I took a slightly different approach with this which accomplishes the same thing.

**Review by trask — APPROVED** [review]

**Pull request merged**